### PR TITLE
Host ephemeral stub HTTP endpoints over TLS

### PR DESCRIPTION
**Pull request by illicitonion**
Depends on https://github.com/matrix-org/synapse/pull/272

### CODE REVIEW

File: tests/01http-server.pl [hunk 1]
```diff
@@ -5,6 +6,9 @@
 use SyTest::HTTPClient;
 use SyTest::HTTPServer::Request;
 
+require IO::Async::SSL;
```
Comment (leonerd) [inline tests/01http-server.pl]: 'require' is fine here but for consistency with all the others above, might as well 'use' it.
Also needs blank line below.

File: tests/01http-server.pl [hunk 2]
```diff
@@ -92,11 +96,15 @@
             socktype => "stream",
             port     => 0,
          },
+         extensions => ["SSL"],
+         SSL_cert_file => "$DIR/../keys/tls-selfsigned.crt",
+         SSL_key_file => "$DIR/../keys/tls-selfsigned.key",
+
```
Comment (leonerd) [inline tests/01http-server.pl]: Extraneous blank line

Comment (illicitonion) [inline tests/01http-server.pl]: Done

File: tests/01http-server.pl (hunk 1 above)
Comment (illicitonion) [inline tests/01http-server.pl]: Done

**Comment by illicitonion** [comment]
Still blocked on matrix-org/synapse#272

**Comment by leonerd** [comment]
LGTM, pending synapse commit.

**Pull request merged**